.github/workflows/nemo_v5_tests.yml

```diff
@@ -0,0 +1,222 @@
+        # This setting causes the tests to 'fail' if [skip ci] is specified
+        fail-fast: true
+        commit-filter: '[skip ci]'
+    - name: Install dependencies
+      run: |
+        module load python/${PYTHON_VERSION}
+        python -m venv .runner_venv
+        source .runner_venv/bin/activate
+        python -m pip install --upgrade pip
+        # Uncomment the below to use the submodule version of fparser rather
+        # than the latest release from pypi.
+        # pip install external/fparser
+        pip install .
+
+    # PSyclone passthrough for 5.0-beta of NEMO.
+    - name: NEMO 5.0 gfortran passthrough
+      run: |
+        # Set up environment
+        source /apps/spack/psyclone-spack/spack-repo/share/spack/setup-env.sh
+        spack unload && spack load nemo-build-environment%gcc@14
+        source .runner_venv/bin/activate
+        export PSYCLONE_NEMO_DIR=${GITHUB_WORKSPACE}/examples/nemo/scripts
+        export PSYCLONE_HOME=${PWD}/.runner_venv
+        export NEMO_DIR=${HOME}/NEMOv5
+
+        # Set up FCM: PATHs are loaded from SPACK, we only need to set the FCFLAGS
+        cd $NEMO_DIR
+        cp $PSYCLONE_NEMO_DIR/KGOs/arch-linux_spack.fcm arch/arch-linux_spack.fcm
+        export FCFLAGS="-fdefault-real-8 -O2 -fcray-pointer -ffree-line-length-none -g"
+
+        # Clean up and compile
+        ./makenemo -r BENCH -m linux_spack -n BENCH_PASSTHROUGH_GCC clean -y
+        ./makenemo -r BENCH -m linux_spack -n BENCH_PASSTHROUGH_GCC -p ${PSYCLONE_NEMO_DIR}/passthrough.py \
+            del_key "key_xios key_iomput key_top" add_key "key_nosignedzero" -j 4 -v 1
+
+        # Run Orca2
+        cd $NEMO_DIR/tests/BENCH_PASSTHROUGH_GCC/EXP00
+        cp $PSYCLONE_NEMO_DIR/KGOs/namelist_cfg_orca2_like namelist_cfg
+        mpirun -np 4 ./nemo
+        python $PSYCLONE_NEMO_DIR/compare_ouput.py $PSYCLONE_NEMO_DIR/KGOs/run.stat.bench.orca2.4mpi.10steps run.stat
+        cat run.stat
+        export VAR_TIME=$(awk '/ step /{print $3}' timing.output | head -n 1 | sed -e 's/s//')
+        echo "Time-stepping duration = " $VAR_TIME
+
+    - name: NEMO 5.0 nvidia passthrough
+      run: |
+        # Set up environment
+        source /apps/spack/psyclone-spack/spack-repo/share/spack/setup-env.sh
+        spack unload && spack load nemo-build-environment%nvhpc@24.5
+        source .runner_venv/bin/activate
+        export PSYCLONE_NEMO_DIR=${GITHUB_WORKSPACE}/examples/nemo/scripts
+        export PSYCLONE_HOME=${PWD}/.runner_venv
+        export NEMO_DIR=${HOME}/NEMOv5
+        
+        # Set up FCM: PATHs are loaded from SPACK, we only need to set the FCFLAGS
+        # We compile at -O1 to permit comparison of the results.
+        cd $NEMO_DIR
+        cp $PSYCLONE_NEMO_DIR/KGOs/arch-linux_spack.fcm arch/arch-linux_spack.fcm
+        export FCFLAGS="-i4 -Mr8 -O1 -Minline -Mcray=pointer -Mpre -g"
+        
+        # Clean up and compile
+        # Without key_mpi_off it fails to compile (even without psyclone)
```

Comment:
Yeah (the mpirun below was the mistake, it ran 4 jobs without mpi), the issue is not produced by psyclone. EWithout psyclone I also get:
```
NVFORTRAN-W-0189-Argument number 1 to mpi_bcast: association of scalar actual argument to array dummy argument (/home/sergi/workspace/psyclone/nemo/nemo-upstream/tests/BENCH_nvhpc_base/BLD/ppsrc/nemo/lib_mpp.f90: 545)
  0 inform,   1 warnings,   0 severes, 0 fatal for mpp_bcast_nml
NVFORTRAN-W-0189-Argument number 1 to mpi_allgather: association of scalar actual argument to array dummy argument (/home/sergi/workspace/psyclone/nemo/nemo-upstream/tests/BENCH_nvhpc_base/BLD/ppsrc/nemo/lib_mpp.f90: 2118)
  0 inform,   1 warnings,   0 severes, 0 fatal for mpp_ini_znl
nvfortran-Fatal-/home/sergi/.psyclone-spack/spack-repo/opt/spack/linux-ubuntu22.04-skylake/gcc-14.1.0/nvhpc-24.5-bvyzfwfhuhq7xvflri7bradlhe5q4vox/Linux_x86_64/24.5/compilers/bin/tools/fort2 TERMINATED by signal 1
```
I believe it is an issue of using OpenMPI instead of the MPIX that comes with nvhpc, but I don't want to delay this PR more, we could look at this in a follow up.